cmds/webboot/utils.go

```diff
@@ -51,17 +53,35 @@ func download(URL, fPath string, uiEvents <-chan ui.Event) error {
 	}
 	defer resp.Body.Close()
 
-	f, err := os.Create(fPath)
+	tempFile, err := ioutil.TempFile("", "iso-download-")
 	if err != nil {
 		return err
 	}
-	defer f.Close()
+	defer os.Remove(tempFile.Name())
 
 	go listenForCancel(ctx, cancel, uiEvents)
 	counter := NewWriteCounter(resp.ContentLength)
-	defer counter.Close()
 
-	if _, err = io.Copy(f, io.TeeReader(resp.Body, &counter)); err != nil {
+	if _, err = io.Copy(tempFile, io.TeeReader(resp.Body, &counter)); err != nil {
+		counter.Close()
+		return err
+	}
+
+	counter.Close()
+	copyProgress := menu.NewProgress("Download complete. Writing ISO to cache", true)
+	defer copyProgress.Close()
+
+	if _, err = tempFile.Seek(0, 0); err != nil {
+		return err
+	}
+
+	cacheFile, err := os.Create(fPath)
+	if err != nil {
+		return err
+	}
+	defer cacheFile.Close()
+
+	if _, err := io.Copy(cacheFile, tempFile); err != nil {
```

Comment:
If this fails will it leave a partial file on the stick? If it fails should we remove the file?

Reply:
Right now it would leave a partial file on disk. Maybe I can add a line to the error case to remove the file?

Reply:
yes sounds good

Reply:
I was thinking of adding `os.Remove(cacheFile.Name())` inside the error case, but would that cause any problems with the deferred `cacheFile.Close()`?

Reply:
no. In fact let's talk about why tomorrow, the reason is interesting.

Reply:
Sounds good, I added the `os.Remove()` line.